controllers/cf/cf_pod_spec.go

```diff
@@ -0,0 +1,318 @@
+		}
+		// containers[*].name
+		if baseContainer.Name != "" {
+			return nil, NewConfigError("containers[0].name")
+		}
+		baseContainer.Name = factoryContainer.Name
+		// containers[*].ports
+		if len(baseContainer.Ports) > 0 {
+			return nil, NewConfigError("containers[0].ports")
+		}
+		baseContainer.Ports = currentContainer.Ports
+		// containers[*].readinessProbe
+		if baseContainer.ReadinessProbe == nil {
+			baseContainer.ReadinessProbe = factoryContainer.ReadinessProbe
+		}
+		// containers[*].resources
+		if len(baseContainer.Resources.Limits) == 0 &&
+			len(baseContainer.Resources.Requests) == 0 {
+			baseContainer.Resources = factoryContainer.Resources
+		}
+		// containers[*].workingDir
+		if baseContainer.WorkingDir != "" {
+			return nil, NewConfigError("containers[0].workingDir")
+		}
+		baseContainer.WorkingDir = currentContainer.WorkingDir
+	} else {
+		base.Containers = make([]ar.ApicurioRegistrySpecDeploymentPodSpecPreviewContainer, len(factory.Containers))
+		for i, c := range factory.Containers {
+			cc, err := ConvertFromContainer(&c)
+			if err != nil {
+				return nil, err
+			}
+			base.Containers[i] = *cc
+		}
+	}
+	// ephemeralContainers
+	if len(base.EphemeralContainers) > 0 {
+		return nil, NewConfigError("ephemeralContainers")
+	}
+	base.EphemeralContainers = current.EphemeralContainers
+	// imagePullSecrets
+	if len(base.ImagePullSecrets) > 0 {
+		return nil, NewConfigError("imagePullSecrets")
+	}
+	base.ImagePullSecrets = current.ImagePullSecrets
+	// initContainers
+	if len(base.InitContainers) > 0 {
+		return nil, NewConfigError("initContainers")
+	}
+	base.InitContainers = current.InitContainers
+	// terminationGracePeriodSeconds
+	if base.TerminationGracePeriodSeconds == nil {
+		base.TerminationGracePeriodSeconds = factory.TerminationGracePeriodSeconds
+	}
+	// tolerations
+	if len(base.Tolerations) > 0 {
+		return nil, NewConfigError("tolerations")
+	}
+	base.Tolerations = current.Tolerations
+	// volumes
+	for _, v := range factory.Volumes {
+		common.SetVolume(log, &base.Volumes, &v)
+	}
+	return base, nil
+}
+
+type SanitizeError struct {
+	isConfigError   bool
+	conversionError error
+	message         string
+}
+
+func (this *SanitizeError) Error() string {
+	return this.message
+}
+
+func NewConfigError(field string) *SanitizeError {
+	return &SanitizeError{
+		isConfigError:   true,
+		conversionError: nil,
+		message:         "field " + field + " is reserved and must not be defined",
+	}
+}
+
+func NewConvertError(err error) *SanitizeError {
+	return &SanitizeError{
+		isConfigError:   false,
+		conversionError: err,
+		message:         err.Error(),
+	}
+}
+
+// Do a magic using JSON to conver these values.
+// They MUST have the equivalent field names.
+// Remember, we are only doing this because we don't have ommitempty tags in PodSpec.
+func ConvertToPodSpec(source *ar.ApicurioRegistrySpecDeploymentPodSpecPreview) (*core.PodSpec, *SanitizeError) {
```

Comment:
I'm really surprised that we should define ser/deser for this manually, but maybe I'm missing something ...